### ISSUE TITLE
Domain / Quality Classifiers failing on nightly

### ISSUE THREAD

**Issue by praateekmahajan** [NVIDIA/NeMo-Curator#419]
**Describe the bug**

It was working fine until a few weeks ago while now there might've been a way in how model is being persisted locally or some upstream HF code change

**Steps/Code to reproduce bug**
```
(rapids) root@39852f30f91b:/# domain_classifier_inference --input-data-dir /data/clean/tinystories_val --output-data-dir /logs/domain_tinystories_val
Arguments parsed = Namespace(device='cpu', files_per_partition=2, n_workers=256, num_files=None, nvlink_only=False, protocol='tcp', rmm_pool_size='512MB', scheduler_address=None, scheduler_file=None, threads_per_worker=1, enable_spilling=False, set_torch_to_use_rmm=False, max_mem_gb_classifier=None, input_data_dir='/data/clean/tinystories_val', output_data_dir='/logs/domain_tinystories_val', input_file_type='jsonl', input_file_extension=None, output_file_type='jsonl', input_text_field='text', batch_size=64, pretrained_model_name_or_path=None, autocast=True, max_chars=2000)
Starting domain classifier inference
Total input files 3
config.json: 100%|████████████████████████████████████████████████| 1.61k/1.61k [00:00<00:00, 12.5MB/s]
config.json: 100%|████████████████████████████████████████████████████| 579/579 [00:00<00:00, 5.41MB/s]
pytorch_model.bin: 100%|█████████████████████████████████████████████| 371M/371M [00:03<00:00, 116MB/s]
model.safetensors: 100%|████████████████████████████████████████████| 371M/371M [00:05<00:00, 73.6MB/s]
model.safetensors: 100%|████████████████████████████████████████████| 735M/735M [00:17<00:00, 42.4MB/s]
tokenizer_config.json: 100%|██████████████████████████████████████| 1.26k/1.26k [00:00<00:00, 10.1MB/s]
spm.model: 100%|██████████████████████████████████████████████████| 2.46M/2.46M [00:00<00:00, 43.3MB/s]
tokenizer.json: 100%|█████████████████████████████████████████████| 8.66M/8.66M [00:00<00:00, 37.1MB/s]
added_tokens.json: 100%|█████████████████████████████████████████████| 23.0/23.0 [00:00<00:00, 263kB/s]
special_tokens_map.json: 100%|████████████████████████████████████████| 286/286 [00:00<00:00, 3.34MB/s]
Fitting memory estimate curve for model: microsoft/deberta-v3-base
Batch size: 100%|████████████████████████████████████████████████████████| 8/8 [01:24<00:00, 10.55s/it]
File Batch ID 0: total input files 3
Reading 3 files
Starting domain classifier inference
Total input Dask DataFrame partitions 3
GPU: tcp://127.0.0.1:39525, Part: 1:   0%|                                   | 0/10000 [00:00<?, ?it/s]
2024-12-10 18:41:29,241 - distributed.worker - ERROR - Compute Failed
Key:       ('read_single_partition-fused-single_partition_write_with_filename-e36773d7637aa051b2cda1bfc7c080bc', 2)
State:     executing
Task:  <Task ('read_single_partition-fused-single_partition_write_with_filename-e36773d7637aa051b2cda1bfc7c080bc', 2) _execute_internal_graph(...)>
Exception: 'OSError("Can\'t load the model for \'microsoft/deberta-v3-base\'. If you were trying to load it from \'https://huggingface.co/models\', make sure you don\'t have a local directory with the same name. Otherwise, make sure \'microsoft/deberta-v3-base\' is the correct path to a directory containing a file named pytorch_model.bin, tf_model.h5, model.ckpt or flax_model.msgpack.")'
Traceback: '  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/dask_expr/_expr.py", line 3799, in _execute_internal_graph\n    res = execute_graph(internal_tasks, cache=cache, keys=[outkey])\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/dask/_task_spec.py", line 786, in execute_graph\n    cache[key] = node(cache)\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/dask/_task_spec.py", line 650, in __call__\n    return self.func(*new_argspec, **kwargs)\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/dask/dataframe/core.py", line 7175, in apply_and_enforce\n    df = func(*args, **kwargs)\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/crossfit/op/base.py", line 98, in __call__\n    output = self.call(data, *args, partition_info=partition_info, **kwargs)\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context\n    return func(*args, **kwargs)\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/crossfit/backend/torch/op/base.py", line 86, in call\n    for output in loader.map(self.model.get_model(self.get_worker())):\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/crossfit/backend/torch/model.py", line 72, in get_model\n    self.load_on_worker(worker)\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/crossfit/backend/torch/hf/model.py", line 85, in load_on_worker\n    setattr(worker, self._model_id, self.load_model(device))\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/nemo_curator/classifiers/domain.py", line 59, in load_model\n    model = HFDeberta.from_pretrained(self.config.identifier)\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn\n    return fn(*args, **kwargs)\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/huggingface_hub/hub_mixin.py", line 571, in from_pretrained\n    instance = cls._from_pretrained(\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/huggingface_hub/hub_mixin.py", line 791, in _from_pretrained\n    model = cls(**model_kwargs)\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/nemo_curator/classifiers/base.py", line 90, in __init__\n    self.model = AutoModel.from_pretrained(config["base_model"])\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/transformers/models/auto/auto_factory.py", line 564, in from_pretrained\n    return model_class.from_pretrained(\n  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/transformers/modeling_utils.py", line 3941, in from_pretrained\n    raise EnvironmentError(\n'


  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/dask/base.py", line 660, in compute
    results = schedule(dsk, keys, **kwargs)
  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/dask_expr/_expr.py", line 3799, in _execute_internal_graph
    res = execute_graph(internal_tasks, cache=cache, keys=[outkey])
  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/crossfit/op/base.py", line 98, in __call__
    output = self.call(data, *args, partition_info=partition_info, **kwargs)
  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/crossfit/backend/torch/op/base.py", line 86, in call
    for output in loader.map(self.model.get_model(self.get_worker())):
  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/crossfit/backend/torch/model.py", line 72, in get_model
    self.load_on_worker(worker)
  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/crossfit/backend/torch/hf/model.py", line 85, in load_on_worker
    setattr(worker, self._model_id, self.load_model(device))
  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/nemo_curator/classifiers/domain.py", line 59, in load_model
    model = HFDeberta.from_pretrained(self.config.identifier)
  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn
    return fn(*args, **kwargs)
  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/huggingface_hub/hub_mixin.py", line 571, in from_pretrained
    instance = cls._from_pretrained(
  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/huggingface_hub/hub_mixin.py", line 791, in _from_pretrained
    model = cls(**model_kwargs)
  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/nemo_curator/classifiers/base.py", line 90, in __init__
    self.model = AutoModel.from_pretrained(config["base_model"])
  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/transformers/models/auto/auto_factory.py", line 564, in from_pretrained
    return model_class.from_pretrained(
  File "/opt/conda/envs/rapids/lib/python3.10/site-packages/transformers/modeling_utils.py", line 3941, in from_pretrained
    raise EnvironmentError(
OSError: Can't load the model for 'microsoft/deberta-v3-base'. If you were trying to load it from 'https://huggingface.co/models', make sure you don't have a local directory with the same name. Otherwise, make sure 'microsoft/deberta-v3-base' is the correct path to a directory containing a file named pytorch_model.bin, tf_model.h5, model.ckpt or flax_model.msgpack.
2024-12-10 18:41:29,255 - distributed.worker.state_machine - WARNING - Async instruction for <Task cancelled name="execute(('read_single_partition-fused-single_partition_write_with_filename-e36773d7637aa051b2cda1bfc7c080bc', 0))" coro=<Worker.execute() done, defined at /opt/conda/envs/rapids/lib/python3.10/site-packages/distributed/worker_state_machine.py:3606>> ended with CancelledError
2024-12-10 18:41:29,418 - distributed.worker.state_machine - WARNING - Async instruction for <Task cancelled name="execute(('read_single_partition-fused-single_partition_write_with_filename-e36773d7637aa051b2cda1bfc7c080bc', 1))" coro=<Worker.execute() done, defined at /opt/conda/envs/rapids/lib/python3.10/site-packages/distributed/worker_state_machine.py:3606>> ended with CancelledError
GPU: tcp://127.0.0.1:35161, Part: 2:   0%|                                    | 0/1990 [00:02<?, ?it/s]
GPU: tcp://127.0.0.1:35369, Part: 0:   0%|                                   | 0/10000 [00:01<?, ?it/s]
GPU: tcp://127.0.0.1:39525, Part: 1:   0%|                                   | 0/10000 [00:01<?, ?it/s]
```

**Comment by sarahyurick** [comment]
Thanks, I am able to reproduce the error here: https://github.com/NVIDIA/NeMo-Curator/actions/runs/12267412329/job/34227476052?pr=421. I can start working on this.

**Comment by gayu-thri** [comment]
I faced similar issue. Upvoting this issue.

**Comment by VibhuJawa** [comment]
@gayu-thri ,  This is caused by a release of Transformers (which they have also fixed upstream). For now please use below versions to make it work.  Also CC: @sarahyurick , who triaged this for us and also fixed in https://github.com/NVIDIA/NeMo-Curator/pull/421

``` 

    "transformers>=4.46.3,!=4.47.0",
```

**Comment by sarahyurick** [comment]
Yes, Vibhu is correct. It looks like Transformers just released a new version 3 hours ago, so 4.47.1 should have the fix as well.
